Main Menu and Circle of Trust

chamika opened the pull request:
Developments up to mid week 4
Application Splash Screen
Main Menu
Main Screen
Circle of Trusts
Add/Edit Circle of Trusts numbers

shruti-gupta commented on PCSA/PCSA/AccordionMenuTableViewController.swift:
```diff
@@ -0,0 +1,270 @@
+     
+     - parameter index: The index of the cell to expand.
+     */
+    private func expandItemAtIndex(index : Int, parent: Int) {
```
Are you doing this to expand table view cells?

chamika replied:
Yes. Is something wrong with the implementation?

shruti-gupta commented on PCSA/PCSA/CircleOfTrustEditViewController.swift:
```diff
@@ -0,0 +1,239 @@
+
+class CircleOfTrustEditViewController: UIViewController, CNContactPickerDelegate,UITextFieldDelegate {
+    
+    //MARK: Properties
```
Can we use proper names here instead of numbers?

chamika replied:
I'll changed it to phone numbers.

shruti-gupta commented on PCSA/PCSA/MenuTableViewController.swift:
```diff
@@ -0,0 +1,361 @@
+                        "Safety Plan Worksheet"], title: "Safety Tools"),
+                      Parent(state: .Collapsed, childs: [
+                        "Benefits of Seeking Staff Support",
+                        "Available Services after a Sexual Assault",
```
Use constants.

chamika replied:
OK. I'll changed that.

shruti-gupta commented on PCSA/PCSA/MenuTableViewController.swift:
```diff
@@ -0,0 +1,361 @@
+     
+     - parameter index: The index of the cell to expand.
+     */
+    private func expandItemAtIndex(index : Int, parent: Int) {
```
This functionality is being used at multiple places. Could you keep a single class for expansion functionality?

chamika replied:
I'll refactor the code

shruti-gupta replied:
Did you refactor it?

chamika replied:
There is no other place which uses the same code. Are you suggesting to move UITableView related code to a single class?

chamika commented:
Changed variable name from numbers to phoneNumbers
Use constants for hard coded String values
Changed App Icon to colored 
Added Main menu application logo
Refer commit: https://github.com/systers/PCSA-iOS/pull/1/commits/299c41ad4bbe0a52c8d9e36754e1e8ca05624621

shruti-gupta commented on PCSA/PCSA/AfterAssualtViewController.swift:
```diff
@@ -0,0 +1,34 @@
+        super.viewDidLoad()
+    }
+
+    override func didReceiveMemoryWarning() {
```
Remove the methods where you are not changing anything.

chamika replied:
Removed unnecessary method overrides

shruti-gupta commented on PCSA/PCSA/Assets.xcassets/AppIcon.appiconset/Contents.json:
```diff
@@ -0,0 +1,78 @@
+    {
+      "size" : "29x29",
+      "idiom" : "iphone",
+      "filename" : "Icon-58.png",
```
Naming convention for image files should be something like: fileName.png, fileName_2x.png, fileName_3x.png

chamika replied:
How can we manage if more than one 2x images exist? These images are different in idioms. Therefore multiple 2x images can exist.

shruti-gupta commented on PCSA/PCSA/BorderedView.swift:
```diff
@@ -0,0 +1,31 @@
+//  Created by Chamika Weerasinghe on 6/26/16.
+//  Copyright © 2016 Peacecorps. All rights reserved.
+//
+
```
Please add class descriptions.

chamika replied:
Added class level descriptions

shruti-gupta commented on PCSA/PCSA/CircleOfTrustEditViewController.swift:
```diff
@@ -0,0 +1,239 @@
+    }
+    
+    //MARK: Actions
+    @IBAction func selectContact1(sender: AnyObject) {
```
can we have tags for the textfields, and have 1 method instead of 5?

chamika replied:
Use UIView tags and refactored it to single function

shruti-gupta commented on PCSA/PCSA/CircleOfTrustEditViewController.swift:
```diff
@@ -0,0 +1,239 @@
+        
+        let saveSuccess = NSKeyedArchiver.archiveRootObject(numbers, toFile: CircleOfTrustEditViewController.ArchiveURL.path!)
+        if(saveSuccess){
+//            UIUtil.showAlert(self, title: "Save", message: "Numbers saved sucessfully", actions:nil)
```
Please remove unused comments

chamika replied:
Removed all unused comments

shruti-gupta commented on PCSA/PCSA/MainViewController.swift:
```diff
@@ -0,0 +1,51 @@
+            menuButton.target = revealViewController()
+            menuButton.action = #selector(SWRevealViewController.revealToggle(_:))
+            view.addGestureRecognizer(self.revealViewController().panGestureRecognizer())
+//            revealViewController().rearViewRevealWidth = 100
```
remove unused code.

chamika replied:
Removed all the unused code
